src/index.js

```diff
@@ -1,3 +1,4 @@
+// @flow
```

Comment:
🤔: From looking at `recompose`, it seems like they're using a different strategy to add typings.  They have a [`types`](https://github.com/acdlite/recompose/tree/master/types) where all the actual `type` declarations live, and they don't have any types directly in their `src` directory.

I'm wondering how the types we declare in-file like this will work if you're a consumer of the library. Did you run into any documentation about how to structure things from a library use case vs an application use case?